controllers/managedosversionchannel_controller.go

```diff
@@ -247,18 +251,20 @@ func (r *ManagedOSVersionChannelReconciler) handleSyncPod(ctx context.Context, p
 		})
 		now := metav1.Now()
 		ch.Status.LastSyncedTime = &now
-		return ctrl.Result{RequeueAfter: interval}, nil
+		return ctrl.Result{RequeueAfter: interval}
 	default:
 		// Any other phase (failed or unknown) is considered an error
-		return ctrl.Result{}, fmt.Errorf("synchronization pod failed")
+		err = fmt.Errorf("synchronization pod failed")
+		return ctrl.Result{RequeueAfter: interval}
 	}
 }
 
-// deletePodOnError deletes the pod if err is not nil
-func (r *ManagedOSVersionChannelReconciler) deletePodOnError(ctx context.Context, pod *corev1.Pod, ch *elementalv1.ManagedOSVersionChannel, err error) {
-	logger := ctrl.LoggerFrom(ctx)
-
+// handleErrors deletes the pod, produces error log traces and sets the failure state if error is not nil, otherwise sets a success state
+func (r *ManagedOSVersionChannelReconciler) handleFailedSync(ctx context.Context, pod *corev1.Pod, ch *elementalv1.ManagedOSVersionChannel, err error) {
 	if err != nil {
+		logger := ctrl.LoggerFrom(ctx)
+		now := metav1.Now()
+		ch.Status.LastSyncedTime = &now
```

Comment:
So we essentially log the error, set lastsynctime to current time, so it also tracks the time it failed.